### PR TITLE
pass headers to coder api

### PR DESCRIPTION
**Pull request by Liorba**
Hey coder team this pr tries to tackle issue #76
This pr added the following functionality to the vscode extension:
1. new set of configuration to set a VPN header 
2. option to set the token directly in `settings.json` or in file 
3. if header name is set and non of the option above was not provided a user will be presented by prompt widow to add his token (similar to session token)
4. if a valid header it will be added to `Axios` global headers.
5. To deal with a case that the provided token expired of invalid after some time, I've added new Axios interceptor. which will check a set of conditions and prompt the user to put new case it discover a bad response due to invalid token
6. pass the vpn header name and token to the SSH config `proxyCommand` 

There is an issue with the coder `vpnssh` command that unlike the `ssh` command do not pass the headers correctly to the underline ssh command. when I replace it with the "regular" ssh command it worked. any Idea what might causing this?

### CODE REVIEW

**Comment by kylecarbs** [comment]
@Liorba I fixed the header problem in the attached PR: https://github.com/coder/coder/pull/7181

I'll review the rest shortly!

File: src/commands.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import axios from "axios"
+import axios  from "axios"
```
Comment (deansheather) [inline src/commands.ts]:
```suggestion
import axios from "axios"
```

File: src/commands.ts [hunk 2]
```diff
@@ -8,8 +8,8 @@ import { Storage } from "./storage"
 import { WorkspaceTreeItem } from "./workspacesProvider"
 
 export class Commands {
-  public constructor(private readonly vscodeProposed: typeof vscode, private readonly storage: Storage) {}
 
+  public constructor(private readonly vscodeProposed: typeof vscode, private readonly storage: Storage) {}
```
Comment (deansheather) [inline src/commands.ts]:
```suggestion
  public constructor(private readonly vscodeProposed: typeof vscode, private readonly storage: Storage) {}

```

**Review by deansheather — COMMENTED** [review]
This should probably just be a map of header name to string or file path so people can set multiple headers

**Comment by Liorba** [comment]
@deansheather,I thought about this approach at first,  but this is no regular header. it sort of like the session token which is a header but has a special treatments.

**Comment by kylecarbs** [comment]
@rodrimaia let's make this more generic with `coder.headers` instead to allow for a series of arbitrary headers!

**Comment by bpmct** [comment]
We did this in https://github.com/coder/vscode-coder/pull/119!

**Pull request closed without merging**